src/wolfgame.py

```diff
@@ -3169,18 +3133,19 @@ def leave(var, what, user, why=None):
     if var.PHASE == "none":
         return
 
+    ps = get_players()
     # Only mark living players as disconnected, unless they were kicked
-    if user.nick in var.PLAYERS and (what == "kick" or user.nick in list_players()): # FIXME: Need to fix this once var.PLAYERS and list_players() hold User instances
+    if user in var.ALL_PLAYERS and (what == "kick" or user in ps):
         # Prevent duplicate entry in var.ORIGINAL_ROLES
         for roleset in var.ORIGINAL_ROLES.values():
             if user.nick in roleset: # FIXME: Need to fix this once the role sets hold User instances
                 roleset.remove(user.nick)
                 roleset.add("(dced)" + user.nick) # FIXME: Need to get rid of all the (dced) hacks
                 break
 
-        var.DCED_PLAYERS[user.nick] = var.PLAYERS.pop(user.nick) # FIXME: Need to fix this once var.{DCED_}PLAYERS hold User instances
+        var.DCED_PLAYERS[user.nick] = var.PLAYERS.pop(user.nick) # FIXME: Need to fix this once var.DCED_PLAYERS hold User instances
```

Comment:
Comment was actually accurate, it refers to both `var.PLAYERS` and `var.DCED_PLAYERS`. Admittedly, that isn't as obvious as I originally thought.